RecoLuminosity/LumiProducer/test/analysis/plugins/PCCNTupler.cc

```diff
@@ -0,0 +1,283 @@
+
+    pileup = fs->make<TH1F>("pileup","pileup",100,0,100);
+    if(includeVertexInformation){
+        tree->Branch("nGoodVtx","map<int,int>",&nGoodVtx); 
+        tree->Branch("num_Trks",&nTrk,"nTrk/I");
+        recoVtxToken=consumes<reco::VertexCollection>(fPrimaryVertexCollectionLabel);
+    }
+
+    if(includePixels){
+        tree->Branch("BXNo","map<int,int>",&BXNo);
+        tree->Branch("nPixelClusters","map<int,int>",&nPixelClusters);
+        tree->Branch("nClusters","map<int,int>",&nClusters);
+        tree->Branch("layers","map<int,int>",&layers);
+        // dead modules
+        nDeadModules = 6;
+        nDeadPrint = 0;
+        deadModules[0] = 302125076; 
+        deadModules[1] = 302125060;
+        deadModules[2] = 302197516;
+        deadModules[3] = 344019460;
+        deadModules[4] = 344019464;
+        deadModules[5] = 344019468;  
+        pixelToken=consumes<edmNew::DetSetVector<SiPixelCluster> >(fPixelClusterLabel);
+    }
+
+}
+
+// ----------------------------------------------------------------------
+PCCNTupler::~PCCNTupler() { }  
+
+// ----------------------------------------------------------------------
+void PCCNTupler::endJob() { 
+    cout << "==>PCCNTupler> Succesfully gracefully ended job" << endl;
+}
+
+// ----------------------------------------------------------------------
+void PCCNTupler::beginJob() {
+
+}
+
+
+void PCCNTupler::beginLuminosityBlock(edm::LuminosityBlock const& lumi, edm::EventSetup const& isetup){
+    firstEvent = true;
+}
+
+void PCCNTupler::endLuminosityBlock(edm::LuminosityBlock const& lumi, edm::EventSetup const& isetup){
+    tree->Fill();
+}
+
+
+// ----------------------------------------------------------------------
+void PCCNTupler::analyze(const edm::Event& iEvent, 
+            const edm::EventSetup& iSetup)  {
+
+    using namespace edm;
+    using reco::VertexCollection;
+
+    saveAndReset=false;
+    sameEvent = (eventNo==(int)iEvent.id().event());
+    sameLumiNib = true; // FIXME where is this info?
+    sameLumiSect = (LSNo==(int)iEvent.getLuminosityBlock().luminosityBlock());
+    
+    // When arriving at the new LS, LN or event the tree 
+    // must be filled and branches must be reset.
+    // The final entry is saved in the deconstructor.
+    saveAndReset = (saveType=="LumiSect" && !sameLumiSect)
+                || (saveType=="LumiNib" && !sameLumiNib)
+                || (saveType=="Event" && !sameEvent);
+   
+    if(   !saveAndReset && !sameLumiSect
+       && !sameLumiNib  && !sameEvent) {
+        std::cout<<"Diff LS, LN and Event, but not saving/resetting..."<<std::endl;
+    }
+
+    if(saveAndReset){
+        if(!firstEvent) tree->Fill();
+        nVtx = 0;
+        nTrk = 0;
+        nPixelClusters.clear();
+        nClusters.clear();
+        layers.clear();
+        BXNo.clear();
+        nGoodVtx.clear();
+    }
+
+    if(sampleType=="MC"){
+        edm::Handle<std::vector< PileupSummaryInfo> > PupInfo;
```

Comment:
It is only mc. I understood consumes/getByToken would be for data.

On Thursday, April 23, 2015, Slava Krutelyov <notifications@github.com>
wrote:

> In RecoLuminosity/LumiProducer/test/analysis/plugins/PCCNTupler.cc
> <https://github.com/cms-sw/cmssw/pull/8702#discussion_r28944264>:
>
> > +        std::cout<<"Diff LS, LN and Event, but not saving/resetting..."<<std::endl;
> > +    }
> > +
> > +    if(saveAndReset){
> > +        if(!firstEvent) tree->Fill();
> > +        nVtx = 0;
> > +        nTrk = 0;
> > +        nPixelClusters.clear();
> > +        nClusters.clear();
> > +        layers.clear();
> > +        BXNo.clear();
> > +        nGoodVtx.clear();
> > +    }
> > +
> > +    if(sampleType=="MC"){
> > +        edm::Handle<std::vector< PileupSummaryInfo> > PupInfo;
>
> I don't see a "consumes" for this one.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/8702/files#r28944264>.
>